src/test/java/io/cdap/plugin/gcp/gcs/sink/GCSBatchSinkCmekKeyTest.java

```diff
@@ -0,0 +1,155 @@
+    Assume.assumeFalse(String.format(messageTemplate, "bucket path"), bucketPath == null);
+
+    serviceAccountFilePath = System.getProperty("service.account.file");
+    Assume.assumeFalse(String.format(messageTemplate, "service account key file"), serviceAccountFilePath == null);
+
+    serviceAccountKey = new String(Files.readAllBytes(Paths.get(new File(serviceAccountFilePath).getAbsolutePath())),
+                                   StandardCharsets.UTF_8);
+
+  }
+
+  private GCSBatchSink.GCSBatchSinkConfig getConfig(@Nullable String project, @Nullable String path,
+                                                    @Nullable String serviceAccountType,
+                                                    @Nullable String serviceFilePath,
+                                                    @Nullable String serviceAccountJson) throws NoSuchFieldException {
+    GCSBatchSink.GCSBatchSinkConfig gcsBatchSinkConfig = new GCSBatchSink.GCSBatchSinkConfig();
+    FieldSetter
+      .setField(gcsBatchSinkConfig, GCPConfig.class.getDeclaredField("project"), project);
+    FieldSetter
+      .setField(gcsBatchSinkConfig, GCSBatchSink.GCSBatchSinkConfig.class.getDeclaredField("path"), path);
+    FieldSetter.setField(gcsBatchSinkConfig, GCSBatchSink.GCSBatchSinkConfig.class.getDeclaredField("format")
+      , "csv");
+    FieldSetter
+      .setField(gcsBatchSinkConfig, GCPReferenceSinkConfig.class.getDeclaredField("referenceName")
+        , "testref");
+    FieldSetter
+      .setField(gcsBatchSinkConfig, GCPConfig.class.getDeclaredField("serviceAccountType"), serviceAccountType);
+    if (!Strings.isNullOrEmpty(serviceFilePath)) {
+      FieldSetter
+        .setField(gcsBatchSinkConfig, GCPConfig.class.getDeclaredField("serviceFilePath"), serviceFilePath);
+    }
+    if (!Strings.isNullOrEmpty(serviceAccountJson)) {
+      FieldSetter
+        .setField(gcsBatchSinkConfig, GCPConfig.class.getDeclaredField("serviceAccountJson"), serviceAccountJson);
+    }
+    return gcsBatchSinkConfig;
+  }
+
+  @Test
+  public void testServiceAccountPath() throws Exception {
+    GCSBatchSink.GCSBatchSinkConfig config = getConfig(project, bucketPath, "filePath",
+                                                       serviceAccountFilePath, null);
+    testValidCmekKey(config);
+    testInvalidCmekKey(config);
+  }
+
+  @Test
+  public void testServiceAccountJson() throws Exception {
+    GCSBatchSink.GCSBatchSinkConfig config = getConfig(project, bucketPath, "JSON",
+                                                       null, serviceAccountKey);
+    testValidCmekKey(config);
+    testInvalidCmekKey(config);
+  }
+
+  public void testValidCmekKey(GCSBatchSink.GCSBatchSinkConfig config) throws Exception {
+    MockFailureCollector collector = new MockFailureCollector("gcssink");
+    config.validate(collector);
+    FieldSetter
+      .setField(config, GCSBatchSink.GCSBatchSinkConfig.class.getDeclaredField("location"), "us-east1");
```

Comment:
as mentioned above, don't set fields like this. It assumes the class is implemented in a specific way.

Mockito shouldn't be used to test code that lives entirely in this project. It should only be used if you need to mock something that you don't have control over.